src/controllers/controller.doughnut.js

```diff
@@ -239,8 +239,8 @@ module.exports = DatasetController.extend({
 		let startAngle = opts.rotation;
 		let i;
 
-		for (i = 0; i < start + count; ++i) {
-			const arc = arcs[i];
+		for (i = 0; i < start + arcs.length; ++i) {
+			const arc = this._cachedMeta.data[i];
```

Comment:
This one is a bit sketchy. Doughnut could use some work on it through, so I guess this is ok for now.

Reply:
yeah, I debated whether it was the best thing to do. I just pushed an update to this part. let me know if you think it's any better